### ISSUE TITLE
Add configurable smoothing/averaging/sampling to logs to limit amount of data written

### ISSUE THREAD

**Issue by hobinjk** [mozilla-iot/gateway#2082]
Something like "sample rate: 1min" would be good. I'm not sure what would be the most user-friendly options here, i.e. if people would want average over that minute, all data to be dropped for 1 minute, or another variation.

**Comment by tim-hellhake** [comment]
Sound a bit like solving problems which are already solved.
Timeseries databases like influxDB allow you to [downsample](https://docs.influxdata.com/influxdb/v1.7/guides/downsampling_and_retention/) older data to save space.

**Comment by hobinjk** [comment]
Yes, this is absolutely intended to be reinventing the wheel. One of the priorities of our current logging system is to not add dependencies to cut down base image size (e.g. influxDB is 100MB)  and keep our build process simple

**Comment by flatsiedatsie** [comment]
It sounds to me like the Gateway needs a general settings page.

**Comment by hobinjk** [comment]
Yes, definitely! But in this case I would expect the setting to be per-log since the retention window is already an equivalent setting done for each log. I also think that while someone might not care about capturing the temperature more than once an hour, that same person might want to have second-resolution information about power fluctuations.
